CoreFoundation/cmake/modules/CoreFoundationAddFramework.cmake

```diff
@@ -22,24 +22,32 @@ function(add_framework NAME)
          NO_SOURCE_PERMISSIONS)
   endif()
   if(AF_PUBLIC_HEADERS)
-    file(COPY
-           ${AF_PUBLIC_HEADERS}
-         DESTINATION
-           ${CMAKE_BINARY_DIR}/${NAME}.framework/Headers
-         NO_SOURCE_PERMISSIONS)
+    foreach(HEADER IN LISTS AF_PUBLIC_HEADERS)
+      get_filename_component(HEADER_FILENAME ${HEADER} NAME)
+      set(DEST ${CMAKE_BINARY_DIR}/${NAME}.framework/Headers/${HEADER_FILENAME})
+      add_custom_command(OUTPUT ${DEST}
+                         DEPENDS ${HEADER}
+                         WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+                         COMMAND ${CMAKE_COMMAND} -E copy ${HEADER} ${DEST})
```

Comment:
This will result in an always out of date build and breaks the null build.  I think that you can swap it out with `copy_if_different` and that *might* be sufficient to fix that at least.

Reply:
I think the `OUTPUT` and `DEPENDS` take care of not executing the command if it is not necessary.

```shell
$ cmake --build . -- clean
[1/1] Cleaning all built files...
Cleaning... 214 files.
$ cmake --build . -- CoreFoundation
... some warnings during the build here …
[214/214] Linking C static library CoreFoundation/libCoreFoundation.a
$ cmake --build . -- -v -d explain CoreFoundation
ninja: no work to do.
$ touch ${CMAKE_SOURCE_DIR}/Base.subproj/ForSwiftFoundationOnly.h
$ cmake --build . -- -v -d explain CoreFoundation
ninja explain: restat of output CoreFoundation.framework/Headers/ForSwiftFoundationOnly.h older than most recent input /home/danielrodriguez/scratch/swift/swift-corelibs-foundation/CoreFoundation/Base.subproj/ForSwiftFoundationOnly.h (1562634736 vs 1562634829)
ninja explain: CoreFoundation.framework/Headers/ForSwiftFoundationOnly.h is dirty
ninja explain: restat of output CoreFoundation.framework/PrivateHeaders/ForSwiftFoundationOnly.h older than most recent input /home/danielrodriguez/scratch/swift/swift-corelibs-foundation/CoreFoundation/Base.subproj/ForSwiftFoundationOnly.h (1562634736 vs 1562634829)
ninja explain: CoreFoundation.framework/PrivateHeaders/ForSwiftFoundationOnly.h is dirty
ninja explain: CoreFoundation/CMakeFiles/CoreFoundation_POPULATE_HEADERS is dirty
ninja explain: CoreFoundation.framework/Headers/ForSwiftFoundationOnly.h is dirty
ninja explain: CoreFoundation.framework/PrivateHeaders/ForSwiftFoundationOnly.h is dirty
[1/2] cd /home/danielrodriguez/scratch/swift/swift-corelibs-foundation/CoreFoundation && /usr/bin/cmake -E copy Base.subproj/ForSwiftFoundationOnly.h /home/danielrodriguez/scratch/swift/build/my_linux/foundation-linux-x86_64/CoreFoundation.framework/Headers/ForSwiftFoundationOnly.h
[2/2] cd /home/danielrodriguez/scratch/swift/swift-corelibs-foundation/CoreFoundation && /usr/bin/cmake -E copy Base.subproj/ForSwiftFoundationOnly.h /home/danielrodriguez/scratch/swift/build/my_linux/foundation-linux-x86_64/CoreFoundation.framework/PrivateHeaders/ForSwiftFoundationOnly.h
```

I might misunderstand your concern, but it seems to work as intended in the example above.